### PR TITLE
WIP - publish maven local

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
publish artifacts to maven local

## :bulb: Motivation and Context
for testing the SDK as a maven dependency

## :green_heart: How did you test it?
App. sample implements the maven artifact.

How to publish:
./gradlew publishReleasePublicationToMavenLocal

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing

## :crystal_ball: Next steps
add javadocs to maven as well

### CODE REVIEW

File: buildSrc/src/main/java/Config.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ object Config {
     val kotlinStdLib = "stdlib-jdk8"
 
     object BuildPlugins {
-        val androidGradle = "com.android.tools.build:gradle:3.5.1"
+        val androidGradle = "com.android.tools.build:gradle:3.6.0-beta01"
```
Comment (marandaneto) [inline buildSrc/src/main/java/Config.kt]: @bruno-garcia AGP supports maven publish ootb now on its new beta version, so no plugins dependency as before.

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,13 @@ licenses:
   - android-sdk-preview-license-.+
   - android-sdk-license-.+
   - google-gdk-license-.+
+install:
```
Comment (marandaneto) [inline .travis.yml]: this is just a try because 1st build also failed, with missing NDK error

File: sentry-sample/build.gradle.kts [hunk 3]
```diff
@@ -58,7 +58,8 @@ android {
 dependencies {
     implementation(fileTree(mapOf("dir" to "libs", "include" to listOf("*.jar"))))
 
-    implementation(project(":sentry-android"))
+//    implementation(project(":sentry-android"))
+    implementation("io.sentry:sentry-android:2.0.0-SNAPSHOT")
```
Comment (bruno-garcia) [inline sentry-sample/build.gradle.kts]: Does this mean we need to run `release` task before updating the sample?
Or can we just change code and hit 'play' on IntelliJ for example?

File: buildSrc/src/main/java/Config.kt (hunk 1 above)
Comment (bruno-garcia) [reply buildSrc/src/main/java/Config.kt]: nice

File: sentry-sample/build.gradle.kts (hunk 3 above)
Comment (marandaneto) [reply sentry-sample/build.gradle.kts]: @bruno-garcia I've inverted the comment, my bad.

**Comment by marandaneto** [comment]
> LGTM but CI isn't happy yet

NDK not configured. Download it with SDK manager. Preferred NDK version is '20.0.5594570'

maybe because it's still beta?!

File: gradle.properties [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
 
 # Parallel execution
 org.gradle.parallel=true
 
 # Cacheable unit tests
 android.testConfig.useRelativePath = true
+
+# Migrating to AndroidX - AGP 3.6.x
+android.useAndroidX=true
+#android.enableJetifier=true - causing SHA1 digest error
```
Comment (marandaneto) [inline gradle.properties]: similar issue https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/455

File: buildSrc/src/main/java/Config.kt [hunk 5]
```diff
@@ -3,7 +3,7 @@ object Config {
     val kotlinStdLib = "stdlib-jdk8"
 
     object BuildPlugins {
-        val androidGradle = "com.android.tools.build:gradle:3.5.2"
+        val androidGradle = "com.android.tools.build:gradle:3.6.0-beta03"
```
Comment (marandaneto) [inline buildSrc/src/main/java/Config.kt]: @ninniuz what do you think about that?
I'd like to use `maven-publish` with the native support that AGP 3.6.x provides.

Comment (ninniuz) [reply buildSrc/src/main/java/Config.kt]: It is a great addition to AGP and it makes publishing a lot easier, but I wouldn't use beta releases as they could affect aar generation.

File: sentry-android/build.gradle.kts [hunk 6]
```diff
@@ -30,3 +32,13 @@ dependencies {
     api(project(":sentry-android-core"))
     api(project(":sentry-android-ndk"))
 }
+
+afterEvaluate {
```
Comment (ninniuz) [inline sentry-android/build.gradle.kts]: Can you factor out the repeated publishing logic into the main `build.gradle` file?

File: buildSrc/src/main/java/Config.kt (hunk 5 above)
Comment (marandaneto) [reply buildSrc/src/main/java/Config.kt]: @ninniuz based on the changelog, the "only" different thing that affects aar generation would be `useNewApkCreator` which I've disabled, that's why I think it's safe to go, but I do understand your concern.

**Comment by marandaneto** [comment]
for now in favor of https://github.com/getsentry/sentry-android/pull/129

**Comment by marandaneto** [comment]
using https://github.com/novoda/bintray-release for now

**Pull request closed without merging**